ext/standard/array.c

```diff
@@ -5851,34 +5853,38 @@ PHP_FUNCTION(array_rand)
 		 * For N=10 this becomes smaller than 0.1%. */
 		if (HT_IS_PACKED(ht)) {
 			do {
-				zend_long randval = php_mt_rand_range(0, ht->nNumUsed - 1);
-				zval *zv = &ht->arPacked[randval];
+				randval = algo->range(status, 0, ht->nNumUsed - 1);
+				zv = &ht->arPacked[randval];
 				if (!Z_ISUNDEF_P(zv)) {
-					RETURN_LONG(randval);
+					ZVAL_LONG(retval, randval);
+					return true;
 				}
-			} while (1);
+			} while (true);
 		} else {
 			do {
-				zend_long randval = php_mt_rand_range(0, ht->nNumUsed - 1);
-				Bucket *bucket = &ht->arData[randval];
-				if (!Z_ISUNDEF(bucket->val)) {
-					if (bucket->key) {
-						RETURN_STR_COPY(bucket->key);
+				randval = algo->range(status, 0, ht->nNumUsed - 1);
+				b = &ht->arData[randval];
+				if (!Z_ISUNDEF(b->val)) {
+					if (b->key) {
+						ZVAL_STR_COPY(retval, b->key);
 					} else {
-						RETURN_LONG(bucket->h);
+						ZVAL_LONG(retval, b->h);
 					}
+					return true;
 				}
-			} while (1);
+			} while (true);
 		}
 	}
 
 	if (num_req <= 0 || num_req > num_avail) {
-		zend_argument_value_error(2, "must be between 1 and the number of elements in argument #1 ($array)");
-		RETURN_THROWS();
+		if (!silent) {
+			zend_argument_value_error(2, "must be between 1 and the number of elements in argument #1 ($array)");
+		}
+		return false;
 	}
 
-	/* Make the return value an array only if we need to pass back more than one result. */
-	array_init_size(return_value, (uint32_t)num_req);
+	/* Make the return value an array only if we need to pass back more than one result. */	
```

Comment:
A trailing tab was added at the end on this line
(also, several lines were added with trailing tabs or spaces)

Reply:
fixed: 196499ee5c40d15b6ae5f4739787eac10ef02570

Reply:
fixed: 196499ee5c40d15b6ae5f4739787eac10ef02570